src/pages/Content/index.js

```diff
@@ -27,11 +27,51 @@ function sendMessage(message) {
 // The function receives a selector and triggers the click event on it.
 function click(selector) {
   const element = document.querySelector(selector) || null;
-  if (element) element.click();
+  if (element) {
+    try {
+      element.click();
+      return true
+    } catch (e) {
+      throw e;
+    }
+  }
+  else return false;
 }
+
 // The function plays a recording when needed
-function playRecording(recordedEvents) {
-  recordedEvents.forEach(element => {
-    click(element);
-  });
+async function playRecording(recordedEvents) {
+  for (const element of recordedEvents) {
+    try {
+      const status = await tryClickUntilExists(element, 1000, 5);
+      if (!status) {
+        sendMessage({ action: "finishedPlaying", value: `failed on element ${element}` });
+        return;
+      }
+    } catch (e) {
+      sendMessage({ action: "finishedPlaying", value: `error occured ${e}` });
+    }
+  }
+  sendMessage({ action: "finishedPlaying", value: "finished playing" });
+}
+
+//Click function that expects selector, interval to retry, maximum retries as params
+function tryClickUntilExists(selector, interval = 400, maxRetries = 1) {
+  return new Promise((resolve, reject) => {
+    let retries = 1;
+    let status = false;
+    //try until click exists
+    let clickInterval = setInterval(() => {
+      try {
+        status = click(selector);
```

Comment:
you can do here `status = true` right after the `click` invocation. If it reached that line then it didn't throw an exception.

Reply:
Updated the code as advised.